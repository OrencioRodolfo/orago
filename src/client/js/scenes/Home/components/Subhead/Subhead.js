import React from 'react';
import styled from 'styled-components';
import RestricitedContainer from '../../../../components/Containers/Restricited/Restricited';
import SpaceBwContainer from '../../../../components/Containers/SpaceBetween/SpaceBetween';
import Button from '../../../../components/Button/Button';
import PillLeft from '../../../../components/Button/PillLeft/PillLeft';
import PillRight from '../../../../components/Button/PillRight/PillRight';
import Icon from '../../../../components/Icon/Icon';

const Wrapper = styled.div`
  padding: 30px 0;
  background: ${props => props.theme.color('background', 'lighter')};
`;

const Subhead = () => (
  <Wrapper>
    <RestricitedContainer>
      <SpaceBwContainer>
        <span>orodolfo / me</span>
        <Button>
          <PillLeft>
            <Icon icon="star" />
            Star
          </PillLeft>
          <PillRight>0</PillRight>
        </Button>
      </SpaceBwContainer>
    </RestricitedContainer>
  </Wrapper>
);

export default Subhead;
