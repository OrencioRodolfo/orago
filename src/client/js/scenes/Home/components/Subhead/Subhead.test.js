import React from 'react';
import { shallow } from 'enzyme';
import Subhead from './Subhead';

describe('Subhead component', () => {
  it('Must render properly', () => {
    const wrapper = shallow(
      <Subhead />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
