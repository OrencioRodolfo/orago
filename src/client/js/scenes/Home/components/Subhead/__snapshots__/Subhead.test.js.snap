// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subhead component Must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Subhead />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Container>
        <styled.div>
          <Link
            className="link"
            url="https://github.com/OrencioRodolfo/orodolfo"
          >
            OrencioRodolfo / 
            <b>
              orodolfo
            </b>
          </Link>
          <Button>
            <PillLeft>
              <Icon
                icon="star"
              />
              Star
            </PillLeft>
            <PillRight>
              0
            </PillRight>
          </Button>
        </styled.div>
      </Container>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": <styled.div>
          <Link
            className="link"
            url="https://github.com/OrencioRodolfo/orodolfo"
          >
            OrencioRodolfo / 
            <b>
              orodolfo
            </b>
          </Link>
          <Button>
            <PillLeft>
              <Icon
                icon="star"
              />
              Star
            </PillLeft>
            <PillRight>
              0
            </PillRight>
          </Button>
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              className="link"
              url="https://github.com/OrencioRodolfo/orodolfo"
            >
              OrencioRodolfo / 
              <b>
                orodolfo
              </b>
            </Link>,
            <Button>
              <PillLeft>
                <Icon
                  icon="star"
                />
                Star
              </PillLeft>
              <PillRight>
                0
              </PillRight>
            </Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                "OrencioRodolfo / ",
                <b>
                  orodolfo
                </b>,
              ],
              "className": "link",
              "url": "https://github.com/OrencioRodolfo/orodolfo",
            },
            "ref": null,
            "rendered": Array [
              "OrencioRodolfo / ",
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "orodolfo",
                },
                "ref": null,
                "rendered": "orodolfo",
                "type": "b",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <PillLeft>
                  <Icon
                    icon="star"
                  />
                  Star
                </PillLeft>,
                <PillRight>
                  0
                </PillRight>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <Icon
                      icon="star"
                    />,
                    "Star",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "icon": "star",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "Star",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "children": "0",
                },
                "ref": null,
                "rendered": "0",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Container>
          <styled.div>
            <Link
              className="link"
              url="https://github.com/OrencioRodolfo/orodolfo"
            >
              OrencioRodolfo / 
              <b>
                orodolfo
              </b>
            </Link>
            <Button>
              <PillLeft>
                <Icon
                  icon="star"
                />
                Star
              </PillLeft>
              <PillRight>
                0
              </PillRight>
            </Button>
          </styled.div>
        </Container>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <styled.div>
            <Link
              className="link"
              url="https://github.com/OrencioRodolfo/orodolfo"
            >
              OrencioRodolfo / 
              <b>
                orodolfo
              </b>
            </Link>
            <Button>
              <PillLeft>
                <Icon
                  icon="star"
                />
                Star
              </PillLeft>
              <PillRight>
                0
              </PillRight>
            </Button>
          </styled.div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Link
                className="link"
                url="https://github.com/OrencioRodolfo/orodolfo"
              >
                OrencioRodolfo / 
                <b>
                  orodolfo
                </b>
              </Link>,
              <Button>
                <PillLeft>
                  <Icon
                    icon="star"
                  />
                  Star
                </PillLeft>
                <PillRight>
                  0
                </PillRight>
              </Button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "OrencioRodolfo / ",
                  <b>
                    orodolfo
                  </b>,
                ],
                "className": "link",
                "url": "https://github.com/OrencioRodolfo/orodolfo",
              },
              "ref": null,
              "rendered": Array [
                "OrencioRodolfo / ",
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "orodolfo",
                  },
                  "ref": null,
                  "rendered": "orodolfo",
                  "type": "b",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <PillLeft>
                    <Icon
                      icon="star"
                    />
                    Star
                  </PillLeft>,
                  <PillRight>
                    0
                  </PillRight>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <Icon
                        icon="star"
                      />,
                      "Star",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "icon": "star",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "Star",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "children": "0",
                  },
                  "ref": null,
                  "rendered": "0",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
